niri-config/src/lib.rs

```diff
@@ -3364,6 +3391,52 @@ where
     }
 }
 
+impl<S: knuffel::traits::ErrorSpan> knuffel::DecodeScalar<S> for ModKey {
```

Comment:
Why do you need a custom DecodeScalar? knuffel can use FromStr on its own with `unwrap(argument, str)`